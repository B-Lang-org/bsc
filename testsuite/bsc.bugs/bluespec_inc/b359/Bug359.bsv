
typedef union tagged {
    void Yellow;
    Bit #(1) Red;
} RgbColorTagged  deriving (Eq, Bits, Bounded);

(* synthesize *)
module mkTestbench ();
     RgbColorTagged color = unpack(0);
     Integer bitty;
     case (pack(color))
       0: bitty = 0;
       default: bitty = 1;
     endcase
     error(integerToString(bitty));
endmodule : mkTestbench
