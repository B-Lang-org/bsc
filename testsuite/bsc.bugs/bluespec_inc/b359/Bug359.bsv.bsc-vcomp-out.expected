checking package dependencies
compiling Bug359.bsv
code generation for mkTestbench starts
Error: "Bug359.bsv", line 15, column 12: (S0015)
  Bluespec evaluation-time error: 1
  During elaboration of `error' at "Bug359.bsv", line 15, column 6.
  During elaboration of `mkTestbench' at "Bug359.bsv", line 8, column 8.
